src/adapter.ts

```diff
@@ -165,30 +196,26 @@ class MongooseAdapter {
    *
    * @param {Boolean} [commit=true] Flag that represents if automatic commit should be enabled or not
    */
-  setAutoCommit (commit = true) {
+  setAutoCommit(commit = true) {
     if (this.isSynced) this.autoCommit = commit;
   }
 
   /**
    * SyncedAdapter: Gets active session or starts a new one. Sessions are used to handle transactions.
    * @returns {Promise<Session>}
    */
-  async getSession () {
+  async getSession() {
     if (this.isSynced) {
-      return this.session && !this.session.hasEnded() ? this.session : this.mongoseInstance.startSession();
+      return this.session && this.session.inTransaction() ? this.session : this.mongoseInstance.startSession();
     } else throw new InvalidAdapterTypeError('Transactions are only supported by SyncedAdapter. See newSyncedAdapter');
   }
 
   /**
    * SyncedAdapter: Sets current session to specific one. Do not use this unless you know what you are doing.
    */
-  async setSession (session) {
+  async setSession(session: ClientSession) {
     if (this.isSynced) {
-      if (session && (session.hasEnded && session.hasEnded.constructor && session.hasEnded.call && session.hasEnded.apply) && !session.hasEnded()) {
-        this.session = session;
-      } else {
-        throw new AdapterError('Tried to set an invalid session');
-      }
+      this.session = session;
```

Comment:
Probably needs checking that `hasEnded()` is not necessary error case at this point.

Reply:
@Zxilly

Reply:
Yes, I will try to test this.

Reply:
It seems that it's an attribute [at least](https://github.com/mongodb/node-mongodb-native/blob/4f9d4ceafdcca01d49e6eeb286a9c27ddf485969/src/sessions.ts#L68).

Reply:
@Sefriol Thanks a lot! Seems this attribute has been removed. I will try to resolve this.